store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreSchemaTest.java

```diff
@@ -70,101 +82,187 @@ public void setUp() throws Exception {
 
-        LIBRARY.addSchema(aSchema1ID, aSchema);
+        fStore.execute(new AddGraph.Builder()
+                .graphId("c")
+                .schema(aSchema)
+                .storeProperties(PROPERTIES)
+                .build(), testContext);
+        // When
+        Collection<String> graphIds = fStore.getAllGraphIds(testUser);
+
+        // Then
+        HashSet<String> expected = new HashSet<>();
+        expected.addAll(Arrays.asList("a", "b", "c"));
+
+        assertTrue(expected.equals(graphIds));
+    }
+
+    @Test
+    public void shouldGetCorrectDefaultViewForAChosenGraphOperation() throws
+            Exception {
+        // Given
+        addGroupCollisionGraphs();
+
+        // When
+        final CloseableIterable<? extends Element> a = fStore.execute(new OperationChain.Builder()
+                .first(new GetAllElements.Builder()
+                        //No view so makes default view, should get only view compatible with graph "a"
+                        .option(FederatedStoreConstants.KEY_OPERATION_OPTIONS_GRAPH_IDS, "a")
+                        .build())
+                .build(), testContext);
+
+        // Then
+        assertNotNull(a);
+        assertFalse(a.iterator().hasNext());
+    }
+
+    @Test
+    public void shouldBeAbleToGetElementsWithOverlappingSchemas() throws
+            OperationException {
+        // Given
+        addOverlappingPropertiesGraphs();
+
+        fStore.execute(new AddElements.Builder()
+                .input(new Edge.Builder()
+                        .group("e1")
+                        .source("source1")
+                        .dest("dest1")
+                        .property("prop1", "value1")
+                        .build())
+                .build(), testContext);
+
+        fStore.execute(new AddElements.Builder()
+                .input(new Edge.Builder()
+                        .group("e1")
+                        .source("source1")
+                        .dest("dest1")
+                        .property("prop1", "value1")
+                        .property("prop2", "value2")
+                        .build())
+                .build(), testContext);
+
+        // When
+        final CloseableIterable<? extends Element> a = fStore.execute(new GetElements.Builder()
+                .input(new EntitySeed("source1"))
+                .view(new View.Builder()
+                        .edge("e1", new ViewElementDefinition.Builder()
+                                .properties("prop2")
+                                .build())
+                        .build())
+                .build(), testContext);
+
+        assertNotNull(a);
+        final ArrayList<Element> results = new ArrayList<>();
+        final CloseableIterator<? extends Element> aIterator = a.iterator();
+        while (aIterator.hasNext()) {
+            Element result = aIterator.next();
+            results.add(result);
+        }
+
+        // Then
+        assertEquals(results.size(), 2);
```

Comment:
I would want more robust testing than if more than how many elements.
You asked for a view of `prop2` from an edge that doesn't have it. I'd like it concreted what it will return and that we are happy with it.